src/cli.rs

```diff
@@ -1,3 +1,233 @@
-pub fn run() {
-    println!("Hello, world!");
+#[cfg(test)]
+use mockall::automock;
+
+const HELP: &str = r#"
+HibernaCLI
+Usage:
+    hibernacli [command] [options]
+
+Commands:
+    help              Display this help message
+    --version or -v   Display the version of the application
+    device [cmd]      Manage devices
+    config [cmd]      Manage configuration
+"#;
+
+const INVALID_COMMAND: &str = "Invalid command, use 'help' to display available commands";
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+trait UserInterface {
+    fn write(&mut self, message: String) -> Result<(), String>;
```

Comment:
You may also want to accept a &str instead of a String here. That would avoid a lot of conversions. You always can pass from String to &str by referencing the string (as long as you can borrow it)

But passing from a &str to a String requires taking ownership over the &str, I guess by some form of copy. If you don't need ownership or mutability, &str is best.